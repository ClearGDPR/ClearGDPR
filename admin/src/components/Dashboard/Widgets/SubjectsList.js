import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';

import TableCard from 'components/core/cards/dashboard/TableCard';

const SubjectsList = ({ subjects, isLoading, errorState }) => {
  const header = ['id', 'firstname', 'email', 'consented_on'];
  // TODO: loading?1

  // once we start reaching into data here, this code becomes janky since data is not ours
  const rows = subjects.map(subject => {
    return {
      id: subject.id,
      consented_on: format(new Date(subject.createdAt), 'DD/MM/YYYY'),
      firstname: subject.data.firstname,
      email: subject.data.email
    };
  });
  return (
    <React.Fragment>
      <TableCard header={header} rows={rows} title="Subjects" />
    </React.Fragment>
  );
};

SubjectsList.propTypes = {
  subjects: PropTypes.array,
  isLoading: PropTypes.bool,
  errorState: PropTypes.bool
};

export default SubjectsList;
