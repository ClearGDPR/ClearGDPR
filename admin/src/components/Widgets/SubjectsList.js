import React from 'react';
import PropTypes from 'prop-types';

import TableCard from 'components/core/cards/dashboard/TableCard';

const SubjectsList = ({ subjects, isLoading, errorState }) => {
  const header = ['id', 'name', 'email', 'state'];
  // TODO: loading?
  return (
    <React.Fragment>
      <TableCard header={header} rows={subjects} title="Subjects" />
    </React.Fragment>
  );
};

SubjectsList.propTypes = {
  subjects: PropTypes.array,
  isLoading: PropTypes.bool,
  errorState: PropTypes.bool
};

export default SubjectsList;
