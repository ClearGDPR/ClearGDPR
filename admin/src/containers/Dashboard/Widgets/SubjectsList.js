import React from 'react';
import config from 'config';
import SubjectsListComponent from 'components/Dashboard/Widgets/SubjectsList';
import internalFetch from 'helpers/internal-fetch';
import Paginate from 'components/core/Paginate';

export class SubjectsListContainer extends React.Component {
  state = {
    isLoading: false,
    errorState: false,
    paging: {},
    subjects: []
  };

  fetchSubjects(page = 1) {
    this.setState({ errorState: false, loading: true });
    internalFetch(`${config.API_URL}/api/management/subjects?page=${page}`)
      .then(({ data, paging }) => {
        this.setState({ subjects: data, paging, loading: false });
      })
      .catch(() => {
        this.setState({ errorState: true, loading: false });
      });
  }

  componentDidMount() {
    this.fetchSubjects();
  }

  handlePageClick = data => {
    const page = data.selected;
    this.fetchSubjects(page + 1);
  };

  render() {
    return (
      <div>
        <SubjectsListComponent {...this.state} />
        <Paginate
          pageCount={(this.state.paging && this.state.paging.total) || 1}
          onPageChange={this.handlePageClick}
        />
      </div>
    );
  }
}

export default SubjectsListContainer;
