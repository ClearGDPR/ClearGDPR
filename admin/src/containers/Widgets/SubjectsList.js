import React from 'react';

import SubjectsListComponent from 'components/Widgets/SubjectsList';
import internalFetch from './internal-fetch';

export class SubjectsListContainer extends React.Component {
  state = {
    isLoading: false,
    errorState: false,
    subjects: []
  };

  fetchSubjects() {
    this.setState({ errorState: false, loading: true });
    internalFetch('/api/management/subjects/list')
      .then(({ data: subjects, paging }) => {
        this.setState({ subjects, paging, loading: false });
      })
      .catch(err => {
        this.setState({ errorState: true, loading: false });
      });
  }

  componentDidMount() {
    this.fetchSubjects();
  }

  render() {
    return <SubjectsListComponent {...this.state} />;
  }
}

export default SubjectsListContainer;
