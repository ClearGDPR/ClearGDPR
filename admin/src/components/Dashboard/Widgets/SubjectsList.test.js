import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubjectsList from './SubjectsList';
import { setHours } from 'date-fns';

const setup = propOverrides => {
  const props = Object.assign(
    {
      subjects: [
        {
          id: '0x0',
          data: {
            firstname: 'test',
            email: 'test@test.com'
          },
          consented_on: '2018-07-04T18:33:43.534Z'
        }
      ],
      isLoading: false,
      errorState: false
    },
    propOverrides
  );
  const component = shallow(<SubjectsList {...props} />);

  return { props, component };
};

describe('(Component) SubjectList', () => {
  it('should render', () => {
    const { component } = setup();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render if loading', () => {
    const { component } = setup({ isLoading: true });
    expect(toJson(component)).toMatchSnapshot();
  });
});
