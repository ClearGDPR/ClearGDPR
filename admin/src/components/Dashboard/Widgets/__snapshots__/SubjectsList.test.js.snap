// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SubjectList should render 1`] = `
<React.Fragment>
  <TableCard
    cols={12}
    columns={
      Object {
        "consented_on": Object {
          "title": "Consented",
        },
        "email": Object {
          "title": "Email",
        },
        "firstname": Object {
          "title": "First Name",
        },
        "id": Object {
          "title": "ID",
        },
      }
    }
    rows={
      Array [
        Object {
          "consented_on": "Invalid Date",
          "email": "test@test.com",
          "firstname": "test",
          "id": "0x0",
        },
      ]
    }
    title="Subjects"
  />
</React.Fragment>
`;

exports[`(Component) SubjectList should render if loading 1`] = `
<React.Fragment>
  <TableCard
    cols={12}
    columns={
      Object {
        "consented_on": Object {
          "title": "Consented",
        },
        "email": Object {
          "title": "Email",
        },
        "firstname": Object {
          "title": "First Name",
        },
        "id": Object {
          "title": "ID",
        },
      }
    }
    rows={
      Array [
        Object {
          "consented_on": "Invalid Date",
          "email": "test@test.com",
          "firstname": "test",
          "id": "0x0",
        },
      ]
    }
    title="Subjects"
  />
</React.Fragment>
`;
